Modules/_json.c

```diff
@@ -1762,86 +1736,58 @@ encoder_listencode_list(PyEncoderObject *s, _PyAccu *acc,
 static void
 encoder_dealloc(PyObject *self)
 {
+    PyTypeObject *tp = Py_TYPE(self);
     /* bpo-31095: UnTrack is needed before calling any callbacks */
     PyObject_GC_UnTrack(self);
-    encoder_clear(self);
-    Py_TYPE(self)->tp_free(self);
+    encoder_clear((PyEncoderObject *)self);
+    tp->tp_free(self);
```

Comment:
The current code is just fine, no? I don't see the value of tp. It seems like it comes from a previous change that you reverted.

```suggestion
    Py_TYPE(self)->tp_free(self);
```

Reply:
@vstinner 
`Py_DECREF(tp);`is needed if not test is leaked.
This is why I declared tp ;)

Reply:
```bash
test_json leaked [114, 114, 114] references, sum=342
test_json failed in 35.6 sec
```